pageserver/src/layered_repository.rs

```diff
@@ -972,7 +969,11 @@ impl Timeline for LayeredTimeline {
     ///
     /// Validate lsn against initdb_lsn and latest_gc_cutoff_lsn.
     ///
-    fn check_lsn_is_in_scope(&self, lsn: Lsn) -> Result<()> {
+    fn check_lsn_is_in_scope(
+        &self,
+        lsn: Lsn,
+        latest_gc_cutoff_lsn: &RwLockReadGuard<Lsn>,
+    ) -> Result<()> {
         let initdb_lsn = self.initdb_lsn;
         ensure!(
```

Comment:
I thought about that when I added these checks and decided to keep it just to yield different (more precise) error messages. Keeping one extra Lsn for a timeline is not that much of an overhead. So that's the reason why it is done that way